api/agent/data_access.go

```diff
@@ -117,12 +123,22 @@ func NewDirectDataAccess(ds models.Datastore, ls models.LogStore, mq models.Mess
 	return da
 }
 
-func (da *directDataAccess) GetApp(ctx context.Context, appName string) (*models.App, error) {
-	return da.ds.GetApp(ctx, appName)
+func (da *directDataAccess) GetAppByID(ctx context.Context, appID string) (*models.App, error) {
+	return da.ds.GetAppByID(ctx, appID)
+}
+
+func (da *directDataAccess) GetAppByName(ctx context.Context, appName string) (*models.App, error) {
+	return da.ds.GetAppByName(ctx, appName)
 }
 
-func (da *directDataAccess) GetRoute(ctx context.Context, appName string, routePath string) (*models.Route, error) {
-	return da.ds.GetRoute(ctx, appName, routePath)
+// to make it work for both hybrid and full mode "app" parameter would be app name,
+// for direct access we need app ID hence we need to get app object by app name
+func (da *directDataAccess) GetRoute(ctx context.Context, app string, routePath string) (*models.Route, error) {
+	a, err := da.ds.GetAppByName(ctx, app)
```

Comment:
this one also smells. even though we extract an id, there could be the same path under 2 different apps by id (same security hole as previously mentioned)